langchain/vectorstores/supabase.py

```diff
@@ -0,0 +1,212 @@
+from itertools import repeat
+from typing import Any, Iterable, List, Optional, Tuple, Type, Union
+
+from langchain.docstore.document import Document
+from langchain.embeddings.base import Embeddings
+from langchain.vectorstores.base import VectorStore
+
+
+class SupabaseVectorStore(VectorStore):
+    """VectorStore for a Supabase postgres database. Assumes you have the `pgvector`
+    extension installed and a `match_documents` (or similar) function. For more details:
+    https://js.langchain.com/docs/modules/indexes/vector_stores/integrations/supabase
+
+    You can implement your own `match_documents` function in order to limit the search space
+    to a subset of documents based on your own authorization or business logic.
+
+    Note that the Supabase Python client does not yet support async operations.
+    """
+
+    _client: Any
+    # This is the embedding function. Don't confuse with the embedding vectors.
+    # We should perhaps rename the underlying Embedding base class to EmbeddingFunction or something
+    _embedding: Embeddings
+    table_name: str
+    query_name: str
+
+    def __init__(
+        self,
+        client: Any,
+        embedding: Embeddings,
+        table_name: str,
+        query_name: Union[str, None] = None,
+    ) -> None:
+        """Initialize with supabase client."""
+        try:
+            import supabase
+        except ImportError:
+            raise ValueError(
+                "Could not import supabase python package. "
+                "Please install it with `pip install supabase`."
+            )
+
+        if not isinstance(client, supabase.client.Client):
+            raise ValueError("client should be an instance of supabase.client.Client")
+
+        self._client = client
+        self._embedding: Embeddings = embedding
+        self.table_name = table_name or "documents"
+        self.query_name = query_name or "match_documents"
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict[Any, Any]]] = None,
+    ) -> List[str]:
+        docs = self._texts_to_documents(texts, metadatas)
+
+        vectors = self._embedding.embed_documents(list(texts))
+        return self.add_vectors(vectors, docs)
+
+    @classmethod
+    def from_texts(
+        cls: Type["SupabaseVectorStore"],
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict[Any, Any]]],
```

Comment:
i think adding new explicit kwargs shouldn't be issue for linter, but changing kwarg to arg might. so `metadatas` should have default val `None`